### PR TITLE
AMBARI-24833. Extend cloud log rolling options

### PR DESCRIPTION
**Pull request by oleewere**
# What changes were proposed in this pull request?
Extend rolling options:
- add size format 
- add default rolling strategy

## How was this patch tested?
Waiting for UTs

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-LogSearch-Github-PR-Builder/48/
Test PASSed.

File: ambari-logsearch-logfeeder/src/main/java/org/apache/ambari/logfeeder/conf/CloudStorageDestination.java [hunk 1]
```diff
@@ -19,7 +19,7 @@
 package org.apache.ambari.logfeeder.conf;
 
 public enum CloudStorageDestination {
-  HDFS("hdfs"), S3("s3"), GCS("gcs"), ADLS("adls"), NONE("none");
+  HDFS("hdfs"), S3("s3"), GCS("gcs"), ADLS("adls"), WASB("wasb"), NONE("none");
```
Comment (swagle) [inline ambari-logsearch-logfeeder/src/main/java/org/apache/ambari/logfeeder/conf/CloudStorageDestination.java]: Do we need to add secure mode enums? (adlss, wasbs) are the ones that I know exist.

Comment (oleewere) [reply ambari-logsearch-logfeeder/src/main/java/org/apache/ambari/logfeeder/conf/CloudStorageDestination.java]: it is something for logfeeder level, but as we are not supporting core gcs/wasb yet it won't be needed anyway, default-fs is used, if hdfs client will figure out the right filesystem

Comment (swagle) [reply ambari-logsearch-logfeeder/src/main/java/org/apache/ambari/logfeeder/conf/CloudStorageDestination.java]: Makes sense.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-LogSearch-Github-PR-Builder/50/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-LogSearch-Github-PR-Builder/51/
Test PASSed.

File: ambari-logsearch-logfeeder/src/main/java/org/apache/ambari/logfeeder/conf/LogFeederProps.java [hunk 2]
```diff
@@ -224,7 +224,8 @@
   @LogSearchPropertyDescription(
     name = LogFeederConstants.CLOUD_STORAGE_DESTINATION,
     description = "Type of storage that is the destination for cloud output logs.",
-    examples = {"hdfs", "s3", "gcs", "adls", "none"},
+    examples = {"hdfs", "s3", "gcs", "adls", "wasb", "none"},
```
Comment (kasakrisz) [inline ambari-logsearch-logfeeder/src/main/java/org/apache/ambari/logfeeder/conf/LogFeederProps.java]: Can `default-fs` be set here?

File: ambari-logsearch-logfeeder/src/main/java/org/apache/ambari/logfeeder/conf/output/RolloverConfig.java [hunk 3]
```diff
@@ -49,12 +49,32 @@
...
+    examples = {"50"},
+    defaultValue = "10",
+    sources = {LogFeederConstants.LOGFEEDER_PROPERTIES_FILE}
+  )
+  @Value("${"+ LogFeederConstants.CLOUD_ROLLOVER_MAX_BACKUP_FILES + ":10}")
+  private Integer rolloverMaxBackupFiles;
+
+  @LogSearchPropertyDescription(
+    name = LogFeederConstants.CLOUD_ROLLOVER_THRESHOLD_TIME_SIZE_FORMAT,
+    description = "Rollover cloud log file size format (e.g: KB, MB etc.)",
```
Comment (kasakrisz) [inline ambari-logsearch-logfeeder/src/main/java/org/apache/ambari/logfeeder/conf/output/RolloverConfig.java]: Unit or format?

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-LogSearch-Github-PR-Builder/52/
Test PASSed.

**Pull request merged**